system-tests/lib/cre/features/ocr/ocr.go

```diff
@@ -0,0 +1,528 @@
+			return errors.Wrap(wErr, "failed to find worker nodes")
+		}
+
+		bootstrapNode, isBootstrap := creEnv.DonTopology.BootstrapNode()
+		if !isBootstrap {
+			return errors.New("could not find bootstrap node in topology, exactly one bootstrap node is required")
+		}
+
+		_, ocrPeeringCfg, err := cre.PeeringCfgs(bootstrapNode)
+		if err != nil {
+			return errors.Wrap(err, "failed to get peering configs")
+		}
+
+		jobSpecs = append(jobSpecs, jobs.BootstrapOCR3(bootstrapNode.JobDistributorDetails.NodeID, "ocr3-capability", ocr3CapabilityAddress.Address, chainID))
+
+		for _, workerNode := range workerNodes {
+			evmKey, ok := workerNode.Keys.EVM[chainID]
+			if !ok {
+				return fmt.Errorf("failed to get EVM key (chainID %d, node index %d)", chainID, workerNode.Index)
+			}
+
+			// we need the OCR2 key bundle for the EVM chain, because OCR jobs currently run only on EVM chains
+			evmOCR2KeyBundle, ok := workerNode.Keys.OCR2BundleIDs[chainselectors.FamilyEVM]
+			if !ok {
+				return fmt.Errorf("node %s does not have OCR2 key bundle for evm", workerNode.Name)
+			}
+
+			// we pass here bundles for all chains to enable multi-chain signing
+			jobSpecs = append(jobSpecs, jobs.WorkerOCR3(workerNode.JobDistributorDetails.NodeID, ocr3CapabilityAddress.Address, evmKey.PublicAddress.Hex(), evmOCR2KeyBundle, workerNode.Keys.OCR2BundleIDs, ocrPeeringCfg, chainID))
+			jobSpecs = append(jobSpecs, jobs.DonTimeJob(workerNode.JobDistributorDetails.NodeID, donTimeAddress.Address, evmKey.PublicAddress.Hex(), evmOCR2KeyBundle, ocrPeeringCfg, chainID))
+		}
+	}
+
+	return jobs.Create(ctx, creEnv.CldfEnvironment.Offchain, creEnv.DonTopology, jobSpecs)
+}
+
+// TODO make our DON use clclient from the CTF instead, since then we could remove nodeSetOutput as DON has Nodes that would have this client initialised
+func waitForLogPollerToBeHealthy(dons []*cre.DON, nodeSetOutput []*cre.WrappedNodeOutput) error {
+	for idx, nodeSetOut := range nodeSetOutput {
+		if !dons[idx].HasFlag(cre.ConsensusCapability) || !dons[idx].HasFlag(cre.VaultCapability) {
+			continue
+		}
+		nsClients, cErr := clclient.New(nodeSetOut.CLNodes)
+		if cErr != nil {
+			return errors.Wrap(cErr, "failed to create node set clients")
+		}
+		eg := &errgroup.Group{}
+		for _, c := range nsClients {
+			eg.Go(func() error {
+				return c.WaitHealthy(".*ConfigWatcher", "passing", 100)
+			})
+		}
+		if waitErr := eg.Wait(); waitErr != nil {
+			return errors.Wrap(waitErr, "failed to wait for ConfigWatcher health check")
+		}
+	}
+
+	return nil
+}
+
+/*
+	CODE BELOW WAS COPIED FROM VAROIUS PLACES IN THE SYSTEM TESTS AND (SOMETIMES) MODIFIED
+	TO SHOWCASE THE CONCEPT OF FEATURES.
+
+	IN THE FUTURE, WE SHOULD REFACTOR THE CODE TO AVOID DUPLICATIONS.
+*/
+
+// for now copy from system-tests/lib/cre/contracts/keystone.go
+func deployOCR3Contract(qualifier string, selector uint64, env *cldf.Environment, ds datastore.MutableDataStore) (*ks_contracts_op.DeployOCR3ContractSequenceOutput, error) {
```

Comment:
You can simplify this IMO:
- call the op, read the report and extract the address
then return that and use it